package br.com.caelum.exercicio12;

import br.com.caelum.contas.main.ContaCorrente;

public class TesteErro {

	public static void main(String[] args) {
		System.out.println("inicio do main");
		// try { testando try/catch  em torno do método 1 (exercicio 5)
	      metodo1();
	    //}	catch	(NullPointerException	e)	{
			//System.out.println("erro:	"	+	e);
//}
	      
	      System.out.println("fim do main");
	  }

	  static void metodo1() {
	      System.out.println("inicio do metodo1");
	     // try { testando try/catch  em torno do método 2 (exercicio 4)
	      metodo2();
		//}	catch	(NullPointerException	e)	{
			//System.out.println("erro:	"	+	e);
//}
	      System.out.println("fim do metodo1");
	  }

	  static void metodo2() {
	      System.out.println("inicio do metodo2");
	      ContaCorrente cc = new ContaCorrente();
	      try {   //adicionar try  conforme pedia o exercicio 2
	    	  for (int i = 0; i <= 15; i++) {
	    		  //try{  testando dentro do for conforme exercicio 3
	    	      cc.deposita(i + 1000);
	    	      System.out.println(cc.getSaldo());
	    	      if (i == 5) {
	    	          cc = null;
	    	      }
	    	  }
	    	} catch (NullPointerException e) { // adicionar catch conforme pedia o exercicio 2
	    	  System.out.println("erro: " + e);
	    	}
	}
}

	

